crates/bevy_utils/src/parallel_queue.rs

```diff
@@ -26,12 +31,17 @@ impl<T: Default + Send> Parallel<T> {
     ///
     /// If there is no thread-local value, it will be initialized to it's default.
     pub fn scope<R>(&self, f: impl FnOnce(&mut T) -> R) -> R {
-        let cell = self.locals.get_or_default();
-        let mut value = cell.take();
-        let ret = f(&mut value);
-        cell.set(value);
+        let mut cell = self.locals.get_or_default().borrow_mut();
+        let ret = f(cell.deref_mut());
         ret
     }
+
+    /// Mutably borrows the thread-local value.
+    ///
+    /// If there is no thread-local value, it will be initialized to it's default.
+    pub fn borrow_local_mut(&self) -> RefMut<'_, T> {
```

Comment:
```suggestion
    pub fn borrow_local_mut(&self) -> impl DerefMut<Target = T> + '_ {
```

I'd rather users of this API not be directly reliant on the fact that we return a RefMut. Ideally this would have a named wrapper of some kind.